src/Arcus.Messaging.Health/Tcp/TcpHealthServer.cs

```diff
@@ -0,0 +1,161 @@
+
+            return tcpPort;
+        }
+
+        private static JsonSerializerSettings CreateDefaultSerializerSettings()
+        {
+            var serializingSettings = new JsonSerializerSettings
+            {
+                Formatting = Formatting.None, 
+                NullValueHandling = NullValueHandling.Ignore
+            };
+
+            var enumConverter = new StringEnumConverter { AllowIntegerValues = false };
+            serializingSettings.Converters.Add(enumConverter);
+
+            return serializingSettings;
+        }
+
+        /// <summary>
+        /// Triggered when the application host is ready to start the service.
+        /// </summary>
+        /// <param name="cancellationToken">Indicates that the start process has been aborted.</param>
+        public override Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return Task.FromCanceled(cancellationToken);
+            }
+
+            try
+            {
+                _logger.LogTrace("Starting TCP server...");
+                _listener.Start();
+                _logger.LogInformation("TCP server started!");
+
+                return base.StartAsync(cancellationToken);
+            }
+            catch (Exception exception)
+            {
+                return Task.FromException(exception);
+            }
+        }
+
+        /// <summary>
+        /// This method is called when the <see cref="T:Microsoft.Extensions.Hosting.IHostedService" /> starts. The implementation should return a task that represents
+        /// the lifetime of the long running operation(s) being performed.
+        /// </summary>
+        /// <param name="stoppingToken">Triggered when <see cref="M:Microsoft.Extensions.Hosting.IHostedService.StopAsync(System.Threading.CancellationToken)" /> is called.</param>
+        /// <returns>A <see cref="T:System.Threading.Tasks.Task" /> that represents the long running operations.</returns>
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {
+                _logger.LogInformation("Worker running at: {time}", DateTimeOffset.Now);
+
+                await AcceptConnectionAsync();
+            }
+        }
+
+        private async Task AcceptConnectionAsync()
+        {
+            _logger.LogTrace("Accepting TCP client...");
+            using (TcpClient client = await _listener.AcceptTcpClientAsync())
+            {
+                _logger.LogInformation("TCP client accepted!");
+                using (NetworkStream ns = client.GetStream())
+                {
+                    // TODO: something to serialize as BSON instead?
+                    HealthReport report = await _healthService.CheckHealthAsync();
+                    string serialized = JsonConvert.SerializeObject(report, _serializerSettings);
```

Comment:
Thought, would it make sense to log the health report, or just it's status? 🤔

Reply:
The status is already logged. The entire report (w/ entries and everything) is maybe too much.

Reply:
It's already logged, but only when it was serialized & sent back. Maybe we want to move that up here before we serialize it (I'm very skeptical about waiting that long to log it, but maybe just me)

Reply:
Ah, you're right. It's better to move. Thanks.